### DISCUSSION TITLE
Getting a matchError and no data on teslamate

### DISCUSSION THREAD

**Discussion by amartin916** [teslamate-org/teslamate#3747] [General]
I setup teslamate on unraid and when I look at the logs they just are filled with this error and I dont see any data

`2024-03-19 17:37:18.077 [info] Starting logger for 'Apollo'
2024-03-19 17:37:18.481 car_id=1 [info] Start / :online
2024-03-19 17:37:18.483 [error] GenStateMachine :"1" terminating
** (MatchError) no match of right hand side value: {:error, #Ecto.Changeset<action: :update, changes: %{end_date: ~U[2024-03-19 16:30:40.302000Z]}, errors: [end_date: {"end date must be after start date", [constraint: :check, constraint_name: "positive_duration"]}], data: #TeslaMate.Log.State<>, valid?: false>}
    (teslamate 1.28.4) lib/teslamate/vehicles/vehicle.ex:728: anonymous fn/4 in TeslaMate.Vehicles.Vehicle.handle_event/4
    (ecto_sql 3.10.2) lib/ecto/adapters/sql.ex:1352: anonymous fn/3 in Ecto.Adapters.SQL.checkout_or_transaction/4
    (db_connection 2.6.0) lib/db_connection.ex:1710: DBConnection.run_transaction/4
    (teslamate 1.28.4) lib/teslamate/vehicles/vehicle.ex:723: TeslaMate.Vehicles.Vehicle.handle_event/4
    (stdlib 5.2) gen_statem.erl:1397: :gen_statem.loop_state_callback/11
    (stdlib 5.2) proc_lib.erl:241: :proc_lib.init_p_do_apply/3
2024-03-19 17:37:18.810 car_id=1 [info] Start / :online`

Teslamate version 1.28.4

Not sure what to do to fix it, any help would be appreciated.

**Comment by brianmay** [comment]
Some how either the start_date is invalid or the end_date in invalid. As in it is trying to record an entry where the end date is before the start date.

Reasons for this can be complicated, I hardly understand this. Sometimes it might get times from the Tesla server. Sometimes it might get times from your computer. Other times it might get times from the database.

The first simple thing to check is that your clock on the server is showing the correct time, and has the correct timezone (i.e. the UTC time the server thinks it is should be correct).

**Reply by amartin916** [reply]
Checking the timezone and date on the teslamate docker and pgsql docker container (along with the config for pgsql for the timezone) they are all the same and reported the same time.

**Comment by amartin916** [comment]
Looking at the states table your are correct that it inserted a date in the future, just not sure why, I tweaked the date in the states table and it stopped the error, I guess I will see if it happens again

Edit: Letting the car go from online to asleep the date it put into the states table is wrong. The start date was 2024-03-20 14:45:16.830000 the end date it put in 2024-03-20 23:03:11.746081 (when it should have been) 2024-03-20 15:03:11.746081) Any thoughts on where to look on why ?
